nbcelltests/test.py

```diff
@@ -35,18 +39,18 @@ def run(notebook, rules=None, filename=None, override_kernel_name=""):
 
     # output tests to test file
     with open(name, 'w', encoding='utf-8') as fp:
-        fp.write(BASE.format(override_kernel_name=override_kernel_name, path_to_notebook=notebook, coverage=coverage))
+        fp.write(BASE.format(kernel_name=kernel_name, current_env=current_env, path_to_notebook=notebook, coverage=coverage))
```

Comment:
Why pass `current_env` along here, instead of just doing the exclusion check + collapsing to `kernel_name` ?

Reply:
I want all that logic in one place, and to look as much as possible exactly the same as nbval's (not least because one day I hope to replace it by _using_ nbval for that)

![Screenshot from 2020-06-12 14-08-19](https://user-images.githubusercontent.com/1929/84505756-36b3cd80-acb6-11ea-9213-a9c39c0d0a9e.png)

(Did I understand your question right?)

Reply:
https://github.com/computationalmodelling/nbval/blob/710deaa3c2b0ea847b4041c79a9a3bf92eb88936/nbval/plugin.py#L241-L251

Reply:
I'm not sure if I understand then, so I'll wave it through, and maybe clean it up later if it turns out to be possible.

Reply:
To be clear, that's actually from some hoped future version of nbval (i.e. from a PR to allow specification of kernel name).